### PR TITLE
This Supposed to Be a Simple Test Client Change, You Won't Believe What This Developer Did Next!

### LINKED ISSUE #2650: API tests don't test permissions
### What I'm trying to achieve
Tests suite should be precise. Now all non-user actions are tested using `admin_api_client` fixture which operates on superuser privileges level. This way, we don't test if staff API endpoints contain correct permissions checking.
In the future, it may lead to a potential security issue.

### What I expected to happen
Staff API endpoints tests should use `staff_api_client` fixture and add correct permissions at the beginning of the test. Insufficient permissions test case should be covered as well.

### LINKED ISSUE #3003: Refactor API client in tests
Calling API in tests requires us to write some repetitive code:
```
admin_api_client.post(reverse('api'), {'query': query, 'variables': variables})
```

Both the URL and structure of the data dictionary are the same in every API, so we could adjust the `post` method of our `ApiClient` and have something like:
```
admin_api_client.post(query, variables)
```

We could also experiment with passing permissions to be used for a particular call, e.g.
```
admin_api_client.post(query, variables, permissions=[permission_manage_orders])
```

## TODO
- [x] New method `post_grapqhl` for simplified query posting
- [ ] All tests tested with `staff_api_client` instead of `admin_api_client`
- [ ] `admin_api_client` is removed
- [ ] API-related fixtures should be moved to `tests/api/conftest.py`

### PR DESCRIPTION
**Pull request by Pacu2**
In this PR:
- `post_graphql` was adjusted to automatically test permissions if provided
- Changed admin_api_client to staff_api_client(no more superuser-testing)
- Moved api-related fixtures to tests/api/conftest.py
- Basically, added proper permissions to all API tests

close #3003 
close #2650

### CODE REVIEW

File: tests/api/conftest.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        more suitable for standard GraphQL requests and doesn't mismatch with
+        handling multipart requests in Graphene.
+        """
+        if data:
+            data = json.dumps(data)
+        kwargs['content_type'] = 'application/json'
+        return super().post(API_PATH, data, **kwargs)
+
+    def post_graphql(
+            self, query=None, variables=None, permissions=None, **kwargs):
```
Comment (patrys) [inline tests/api/conftest.py]: How useful is the case where `query` is empty? I'd like to propose an alternative signature:

```python
def post_graphql(self, query, *, variables=None, permissions=None, **kwargs):
```

File: tests/api/conftest.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if data:
+            data = json.dumps(data)
+        kwargs['content_type'] = 'application/json'
+        return super().post(API_PATH, data, **kwargs)
+
+    def post_graphql(self, query, variables=None, permissions=None, **kwargs):
+        """Dedicated helper for posting GraphQL queries.
+
+        Sets the `application/json` content type and json.dumps the variables
+        if present.
```
Comment (NyanKiyoshi) [inline tests/api/conftest.py]: Maybe you could also note/ say that if permission were passed it checks (assert) that the permissions are correctly handled.

Comment (Pacu2) [reply tests/api/conftest.py]: Right, will update the docstring soon

File: tests/api/conftest.py (hunk 1 above)
Comment (patrys) [reply tests/api/conftest.py]: Out of curiosity, why skip the single asterisk in the final implementation?

Comment (Pacu2) [reply tests/api/conftest.py]: Couldn't find a use for it in our current test structure, what's more, it enforced to pass the variables as the keyword adding more boilerplate.

**Pull request merged**